### ISSUE TITLE
Marketplace search: Display featured plugins in Calypso

### ISSUE THREAD

**Issue by gcsecsey** [Automattic/wp-calypso#68084]
This is a follow-up issue to #63167.

Use the new API filters created in #63167 to display the featured section and browse page in Calypso.

> ***Note*** A similar change is implemented for the popular section in #68083

Currently, Calypso only displays featured plugins from WPCOM on this endpoint: https://public-api.wordpress.com/wpcom/v2/plugins/featured This list is maintained manually in 2da8a-pb

WP.org stores the featured flag as a taxonomy, and that's synced to ES. This makes it easy to filter for those plugins in ES.

We should:
- [ ] add the WPCOM featured products set to ES as well
- [ ] update Calypso to fetch the featured products section through the `category` filter

**Comment by gcsecsey** [comment]
While working on #63167 I've investigated the original sorting logic on WP.org. When there's no query is provided, it seems that the sorting is defaulted to the date field:

https://github.com/WordPress/wordpress.org/blob/3818ab62c4d64540294f4a332c4980fcdb4bd911/wordpress.org/public_html/wp-content/plugins/plugin-directory/libs/site-search/jetpack-search.php#L955-L961

**Comment by cpapazoglou** [comment]
The featured plugins are currently set in wpcom statically as you mentioned. I am not sure how we might get the same results by using a taxonomy on data ( .org plugins ) we don't handle.


![Image](https://user-images.githubusercontent.com/12430020/194876332-3c7907a3-3fc4-4775-a299-1ca8a560a35c.png)

Could we perhaps, just request through ES those 6 plugins ( by slug ) in Calypso?

**Comment by gcsecsey** [comment]
> Could we perhaps, just request through ES those 6 plugins ( by slug ) in Calypso?

Yes, for that, we'd need to add a new parameter to the API that'd let us query by slugs, similarly to how we currently pass the `fields[]`. This change would differ from https://github.com/Automattic/wp-calypso/issues/64891 in not returning all the fields for each plugin, just the fields that are needed to render search results.

Then, from Calypso, we just need to list the plugins that we'd like to feature on the section. This would let us keep the configuration of which plugins are currently featured in Calypso, and easy to update.
